File: src/main/Java/boj_step13/boj10814.java
import java.io.*;
import java.util.ArrayList;
import java.util.Comparator;
import java.util.StringTokenizer;

public class boj10814 {
    public static void main(String[] args) throws IOException {
        BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter writer = new BufferedWriter(new OutputStreamWriter(System.out));
        int n = Integer.parseInt(reader.readLine());
        int id = 1;
        ArrayList<MemberList> list = new ArrayList<>();
        while(id <= n) {
            StringTokenizer tokenizer = new StringTokenizer(reader.readLine());
            list.add(new MemberList(id++, Integer.parseInt(tokenizer.nextToken()), tokenizer.nextToken()));
        }
        list.sort(Comparator.comparingInt(MemberList::getAge).thenComparingInt(MemberList::getId));
        for (MemberList it: list) {
            writer.append(it.getAge() + " " + it.getName());
            writer.newLine();
        }
        writer.close();
    }
    private static class MemberList {
        int id, age;
        String name;
        MemberList (int id, int age, String name) {
            this.id = id;
            this.age = age;
            this.name = name;
        }
        int getId() { return id; }
        int getAge() { return age; }
        String getName() { return name; }
    }
}

File: src/main/kotlin/boj_step13/boj10814.kt
import java.io.BufferedReader
import java.io.BufferedWriter
import java.io.InputStreamReader
import java.io.OutputStreamWriter
data class MemberNode (val id: Int, val age: Int, val name: String)
fun main() {
    val r = BufferedReader(InputStreamReader(System.`in`))
    val w = BufferedWriter(OutputStreamWriter(System.out))
    val n = r.readLine().toInt()
    val arr = Array<MemberNode?>(n) { null }
    repeat(n) {
        val (ageS, name) = r.readLine().split(" ").map { it }
        val age = ageS.toInt()
        arr[it] = MemberNode(it, age, name)
    }
    arr.sortWith(compareBy<MemberNode?>{ it!!.age }.thenBy { it!!.id })
    arr.forEach {
        w.write("${it!!.age} ${it!!.name}")
        w.newLine()
    }
    w.close()
}